### ISSUE TITLE
CP (i18n): A few more missing translations in the CP

### ISSUE THREAD

**Issue by helloDanuk** [statamic/cms#1697]
+++ UPDATE 2020-08-06: As of beta.43 a lot has changed. I've updated this list, see below https://github.com/statamic/cms/issues/1697#issuecomment-669897220 +++

## A list of a few more missing translations in the CP 

**Statamic version:** 3.0.0.-beta-26
Fresh install from `statamic/statamic`

* * *

### 1/ Link Existing Field or Fieldset

- Sorry, no matching options
- Button text 'Link'
- or
- Placeholder text 'eg. _hero'

<img width="392" alt="Bildschirmfoto 2020-04-16 um 16 36 26" src="https://user-images.githubusercontent.com/8265902/79471593-1abbe480-8003-11ea-8633-d30ef008fee9.png">

### 2/ Password Reset E-Mail

- Hello
- Regards
- If you're having trouble…

<img width="614" alt="Bildschirmfoto 2020-04-16 um 16 41 07" src="https://user-images.githubusercontent.com/8265902/79472198-d250f680-8003-11ea-844e-dca8b94f92dd.png">

### 3/ Reset password error message

- The password must be at least 8 characters.

<img width="381" alt="Bildschirmfoto 2020-04-16 um 16 20 05" src="https://user-images.githubusercontent.com/8265902/79471564-1394d680-8003-11ea-8ca6-e728fa7db3b1.png">

### 4/ Getting started widget

- Please remember: Statamic is commercial software and you may need… → If there will be a text about the license after the beta period, it would make sense to translate this text.

<img width="923" alt="Bildschirmfoto 2020-04-16 um 17 07 53" src="https://user-images.githubusercontent.com/8265902/79473150-0678e700-8005-11ea-9d04-fe05d05d9351.png">

### 5a/ Permissions

- Title 'Create a Role'
- 'role' in Learn more about… (But actually there should be only one 'Learn more about…' button?)

<img width="984" alt="Bildschirmfoto 2020-04-16 um 17 51 15" src="https://user-images.githubusercontent.com/8265902/79478047-3925de00-800b-11ea-8240-bf9071e6515f.png">

### 5b/ Permissions
- No message helper text in the section 'Updates' (also in the `en` version)
- No message helper text in the section 'User' (also in the `en` version)

<img width="939" alt="Bildschirmfoto 2020-04-16 um 18 24 28" src="https://user-images.githubusercontent.com/8265902/79481610-08947300-8010-11ea-8f20-fbd63245865e.png">
<img width="944" alt="Bildschirmfoto 2020-04-16 um 18 24 34" src="https://user-images.githubusercontent.com/8265902/79481613-09c5a000-8010-11ea-869a-e75e0c7cfdb0.png">
<img width="934" alt="Bildschirmfoto 2020-04-16 um 20 49 09" src="https://user-images.githubusercontent.com/8265902/79499766-c11be000-802b-11ea-8823-a2436cd15a95.png">
<img width="912" alt="Bildschirmfoto 2020-04-16 um 20 49 22" src="https://user-images.githubusercontent.com/8265902/79499947-ff190400-802b-11ea-85e8-9ee194affd4e.png">

- There are description texts for 'Manage publish state' and 'Reorder entries' in permissions.php, but they are not showed in the CP
- I am not sure about the navigation sector. It's singular, but alle other sections (collections, taxonomies…) are in plural?

### 6/ Groups

- Title 'Create a User Group'

<img width="991" alt="Bildschirmfoto 2020-04-17 um 00 16 13" src="https://user-images.githubusercontent.com/8265902/79512006-ce8f9500-8040-11ea-8e36-65681fd17779.png">

- 'User group updated' message 

<img width="357" alt="Bildschirmfoto 2020-04-17 um 00 27 09" src="https://user-images.githubusercontent.com/8265902/79512821-8cffe980-8042-11ea-955a-f3062c0e5de0.png">

### 7a/ Forms

- Title 'Create a Form'

<img width="1017" alt="Bildschirmfoto 2020-04-17 um 00 39 25" src="https://user-images.githubusercontent.com/8265902/79513396-021fee80-8044-11ea-909d-2743bf97483d.png">

### 7b/ Edit Form:

- Button 'Add Email'
- Recipient (To)
- Sender (From)
- Subject
- Template

<img width="967" alt="Bildschirmfoto 2020-04-17 um 01 16 55" src="https://user-images.githubusercontent.com/8265902/79515578-4feb2580-8049-11ea-97f1-a425c9f3bab0.png">

### 8/ i.e. Choose a blueprint

- 0/1 selected

<img width="1268" alt="Bildschirmfoto 2020-04-17 um 00 43 07" src="https://user-images.githubusercontent.com/8265902/79513622-8ffbd980-8044-11ea-9abb-a0cc4db39f55.png">

### 9/ Error message (i.e. when leave required fields blank or use spaces when creating a folder)

- The given data was invalid 

<img width="400" alt="Bildschirmfoto 2020-04-17 um 01 19 14" src="https://user-images.githubusercontent.com/8265902/79515648-7c9f3d00-8049-11ea-8bbe-a6886695c61c.png">

### 10/ Globals

- Title 'Create a Global Set'

<img width="986" alt="Bildschirmfoto 2020-04-17 um 01 27 20" src="https://user-images.githubusercontent.com/8265902/79516100-aad14c80-804a-11ea-85e3-a913ea43c012.png">

The CP makes good progress - it's very intuitive and good looking!

**Comment by ebeauchamps** [comment]
Regarding all "Create a ..." text, you have to manually add them in your .json file because the way the views are configured the translator cannot create automatically those keys. Working fine then (I did that in the French translation)

**Comment by helloDanuk** [comment]
### 11/ Configure global set

- Content Model 
<img width="983" alt="Bildschirmfoto 2020-04-17 um 15 46 59" src="https://user-images.githubusercontent.com/8265902/79576457-9b421a00-80c3-11ea-8b33-e7725a31d697.png">


### 12/ Taxonomy

- Title 'Create a Taxonomy' → or as @ebeauchamps wrote, you could add it manually to the json-file.

### 13/ Collections/Navigations

- 'Parent' field in the sidebar

<img width="1019" alt="Bildschirmfoto 2020-04-17 um 16 40 18" src="https://user-images.githubusercontent.com/8265902/79581143-4eae0d00-80ca-11ea-92ac-ef2e157430d9.png">

**Comment by helloDanuk** [comment]
### 14/ Revisions

- Is it possible somehow to localize the date in the revision history?

<img width="464" alt="Bildschirmfoto 2020-04-17 um 16 58 44" src="https://user-images.githubusercontent.com/8265902/79582954-c3824680-80cc-11ea-9bbb-6ca135d0e2fe.png">

- Flag 'Publish', 'Unpublish' and 'Revision' in the revision history

<img width="390" alt="Bildschirmfoto 2020-04-17 um 17 06 10" src="https://user-images.githubusercontent.com/8265902/79583804-d47f8780-80cd-11ea-93e2-b0ea77641acc.png">
<img width="396" alt="Bildschirmfoto 2020-04-17 um 17 00 35" src="https://user-images.githubusercontent.com/8265902/79583251-2673dd80-80cd-11ea-937e-f727f0a52711.png">

- Publish → 'Choose'

<img width="432" alt="Bildschirmfoto 2020-04-17 um 16 51 15" src="https://user-images.githubusercontent.com/8265902/79582205-bdd83100-80cb-11ea-8af9-25c73fcfb7c7.png">

**Comment by helloDanuk** [comment]
### 15/ If entries can contain links (Toggle in the collection configuration), the link field on the create entry page has some untranslated words

- First Child
- Entry

<img width="518" alt="Bildschirmfoto 2020-04-17 um 17 24 41" src="https://user-images.githubusercontent.com/8265902/79585715-7d2ee680-80d0-11ea-86aa-4a48ff5b1a24.png">


### 16/ Dropdowns are often too narrow for german words

it would look nicer, if the subtitles only use one line. But I am not sure, what would be a good solution in general:

- example 1: I can use 'Blueprint wählen' instead of 'Blueprint auswählen' that's fine
- example 2: 'Verfügbare Spalten' on one line would be great
- example 3: really long 'Untergeordneten Eintrag erstellen' on one line does not make sense from UI perspective.

Maybe some flexibility in width until a certain breakpoint could handle the problem. If a translation needs some more space, the dropdown could get wider. If the translation is really long, a max width would still split up the text on two lines.

<img width="358" alt="Bildschirmfoto 2020-04-17 um 17 35 30" src="https://user-images.githubusercontent.com/8265902/79587146-6093ae00-80d2-11ea-888f-11ef4863e842.png">
<img width="442" alt="Bildschirmfoto 2020-04-17 um 17 32 48" src="https://user-images.githubusercontent.com/8265902/79587150-61c4db00-80d2-11ea-9974-e6c1cdf580aa.png">
<img width="297" alt="Bildschirmfoto 2020-04-17 um 17 53 45" src="https://user-images.githubusercontent.com/8265902/79588593-70ac8d00-80d4-11ea-9a75-956655f24628.png">


### 17/ Sidebar

- 'Sidebar' in live preview
- 'Sidebar' in tab-title (when the screen size is too narrow to place the sidebar on the right side)

<img width="290" alt="Bildschirmfoto 2020-04-17 um 18 13 39" src="https://user-images.githubusercontent.com/8265902/79590581-84a5be00-80d7-11ea-9c4a-0cb5d23aec50.png">
<img width="605" alt="Bildschirmfoto 2020-04-17 um 18 14 36" src="https://user-images.githubusercontent.com/8265902/79590585-853e5480-80d7-11ea-8ab8-e49036e14bbc.png">

**Comment by helloDanuk** [comment]
### 18/ Live Preview

- 'Choose…' in Responsive Dropdown

<img width="413" alt="Bildschirmfoto 2020-04-17 um 20 00 52" src="https://user-images.githubusercontent.com/8265902/79599778-62676c80-80e6-11ea-8050-6059fc442cf9.png">

**Comment by helloDanuk** [comment]
### 19/ Edit Blueprint

- 'Main' Section

<img width="472" alt="Bildschirmfoto 2020-04-17 um 22 51 55" src="https://user-images.githubusercontent.com/8265902/79613226-64d5c080-80fe-11ea-81d6-db79db9f8950.png">

- 'linked fieldset'

<img width="214" alt="Bildschirmfoto 2020-04-17 um 22 56 01" src="https://user-images.githubusercontent.com/8265902/79613367-b3835a80-80fe-11ea-9f76-b4b7c07ec304.png">

### 20/ Add fieldtype

- Fieldtype Category names

<img width="365" alt="Bildschirmfoto 2020-04-17 um 22 58 28" src="https://user-images.githubusercontent.com/8265902/79613580-18d74b80-80ff-11ea-919b-9d5aa92c91f3.png">

- All fieldtype names → But this might be too much…

### 21/ Fieldtype: Conditions

- equals
- not
- contains
- contains_any
- custom

<img width="612" alt="Bildschirmfoto 2020-04-17 um 23 03 43" src="https://user-images.githubusercontent.com/8265902/79614061-0b6e9100-8100-11ea-880d-581daa7af4bd.png">

### 22/ Fieldtype: Validation

- Add rule

<img width="504" alt="Bildschirmfoto 2020-04-17 um 23 04 42" src="https://user-images.githubusercontent.com/8265902/79614154-3c4ec600-8100-11ea-839c-e9063480c764.png">

**Comment by helloDanuk** [comment]
### 23/ Fieldtype: Array

- Label (Optional)

<img width="576" alt="Bildschirmfoto 2020-04-18 um 00 16 43" src="https://user-images.githubusercontent.com/8265902/79622946-f141ac80-8119-11ea-91e2-b39a33042975.png">

### 24/ Fieldtype: Date

- Single
- Range

<img width="489" alt="Bildschirmfoto 2020-04-18 um 02 46 08" src="https://user-images.githubusercontent.com/8265902/79623887-d4f43e80-811e-11ea-8b9c-3e081dca6134.png">

### 25/ Fieldtype: Link

- First Child
- Entry

<img width="305" alt="Bildschirmfoto 2020-04-18 um 03 03 30" src="https://user-images.githubusercontent.com/8265902/79624329-4a610e80-8121-11ea-917f-eb0c05946b75.png">

**Comment by helloDanuk** [comment]
Sorry about all those red circles – they look a bit harsh. I should be gone through pretty much everything now. Although I couldn't check multisite so far.

**Comment by mauriciodulce** [comment]
@helloDanuk I think this about translations is not important, you can still help by completing the translations you need, it is easy. I think most of the languages have been translated by community collaboration.

**Comment by ebeauchamps** [comment]
@mauriciodulce Sorry, I don't get your point. You cannot get the CP fully in your language if you cannot add those missing strings in a language file automatically. And even if you add them manually, they will not be displayed in your language. Most of those above are in minified Javascript files that are changing with each Statamic update.

**Comment by Konafets** [comment]
@helloDanuk How did you get the german version of the password forgot view?

**Comment by helloDanuk** [comment]
@Konafets Great addition of the so far hardwired strings in Blade and VueJs.

I have just updated the `de-CH` translation #1655 PR to beta.29. Primarily changes are: Refine instructions to be less passive or "you" specific.

**About the password forgot view:** I am not sure what exactly do you mean? There are a few files outside of the `cms/ressources/lang` folder.

 If you install statamic with `composer create-project statamic/statamic {change_me} --prefer-dist --stability=dev` you will find the folder `/resources/lang/en` in the root directory. 

I duplicated this folder, named it `de-CH` and translated the strings. You will find the translations [here](https://github.com/helloDanuk/v3-translation).

**Comment by Konafets** [comment]
Hi @helloDanuk,

regarding my question for the forgot view: I meant how got the german version of the page, but figured it out by changing the locale in `config/app.php`.

**Comment by helloDanuk** [comment]
+++ UPDATE +++

As of beta.43 a lot has changed. Many translations have already been implemented or the UI has changed during the beta phase. That's why this issue is no longer up to date and I'm now trying to bring some order into it and update the list.

1/ ✅

2/ ✅

3/ to be done

4/ ✅

5a/ ✅

5b/ to be done

6/ ✅

7a/ ✅

7b/ to be done
- Recipient (To)
- Sender (From)
- Subject

8/ to be done
- selected (also when choosing assets or entries)

9/ ✅

10/ ✅

11/ ✅

12/ ✅

13/ I am not sure, what's a good solution. I can edit all titles in a default blueprint manually though, but a first time statamic user might feel uncomfy when the blueprints have fields with english text… Some of the fields are already translated at the moment (i.e. title, content, sidebar), others are not (i.e. parent, sidebar, author). As for parent and sidebar I can add them manually to the language-json-file, but they are not generated when running `php translator generate…`

14/ to be done
- Publish
- Unpublish
- Revision
- Choose
- as for the date, please take a look at #2053 too

15/ ✅

16/ possible UI issue to be done

17/ Same as 13/

18/ to be done

19/ ✅

20/ to be done…
- Categories: Translation possible, but as of beta.43 when changing the locale to de or de-CH 'Media' and 'Relationship' disappear.
<img width="1197" alt="Bildschirmfoto 2020-08-04 um 17 44 58" src="https://user-images.githubusercontent.com/8265902/89531471-97601200-d7f0-11ea-8990-b469ca52be46.png">

- See #2034 for fieldtype names 

21/ to be done?

22/ ✅

23/ to be done… possible to add manually to the json-file, but does not generate when running `php translator generate…`

24/ ✅

25/ ✅

26/ NEW: Global Set without a blueprint
- Instruction text 
<img width="934" alt="Bildschirmfoto 2020-08-03 um 15 28 37" src="https://user-images.githubusercontent.com/8265902/89531580-beb6df00-d7f0-11ea-9b7e-07f78cf009c7.png">

**Comment by ebeauchamps** [comment]
Great work as usual!

26/ nice catch
20/ same here in French, though I am only getting "Tout" (= Alle), and not Text, nor Media and Relationship. Weird.
14/ I only find the English word in some compiled js files, so we need some help from the Gentlemen here. This one is very important, I don't understand the developers from SPIEGEL did not already ask for it...

Cheers

**Comment by jasonvarga** [comment]
I appreciate you keeping this thread updated, but honestly it's super hard to follow now. Please open a new issue with whatever you believe is outstanding. Thanks!
